crates/iota/src/keytool.rs

```diff
@@ -742,8 +765,17 @@ impl KeyToolCommand {
                 let mut hasher = DefaultHash::default();
                 hasher.update(bcs::to_bytes(&intent_msg)?);
                 let digest = hasher.finalize().digest;
-                let iota_signature =
-                    keystore.sign_secure(&address, &intent_msg.value, intent_msg.intent)?;
+
+                let key = keystore.get_key(&address)?;
+                let iota_signature = match key {
+                    StoredKey::KeyPair(_) => {
+                        keystore.sign_secure(&address, &intent_msg.value, intent_msg.intent)?
+                    }
+                    StoredKey::External { source, .. } => {
+                        bail!("External signing is not supported for source: {source}")
+                    }
+                };
```

Comment:
Same here, isn't the match in sign_secure() for the key type sufficient?

Reply:
See above, this is because I split the PR into 2 and kept the structure prepared to match over the external source